aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -2322,12 +2322,20 @@ _scaled_nvfp4_nvfp4(
           const Tensor& scale_b, const SwizzleType swizzle_b,
           const std::optional<Tensor>& bias,
           const c10::ScalarType out_dtype,
-          const bool single_scale,
-          Tensor& out) {
+          Tensor& out,
+          const std::optional<Tensor>& global_scale_a = std::nullopt,
+          const std::optional<Tensor>& global_scale_b = std::nullopt) {
 #ifdef USE_ROCM
   TORCH_CHECK_NOT_IMPLEMENTED(false, "NVFP4 scaling not supported on ROCM");
 #endif
-  TORCH_CHECK_VALUE(single_scale, "Only single-scaled NVFP4 currently supported");
+  std::optional<Tensor> alpha = std::nullopt;
+  if (global_scale_a.has_value() && global_scale_b.has_value()) {
```

Comment:
should this be scale_a or scale_b?

Reply:
I see - if one global scale is passed, trigger the check for both scales present, as opposed to now, where if you pass a single global scale, it'll silently do nothing with it.